dual.go

```diff
@@ -229,9 +230,10 @@ func dvBind(op Op, inputs []*dualValue) (retVal *dualValue, err error) {
 
 	var ret Value
 	if ret, err = op.Do(vals...); err == nil {
-		retVal = dvUnit(ret)
+		return dvUnit(ret), nil
+	} else {
+		return nil, errors.Wrap(err, opDoFail)
```

Comment:
You are right, I was conflating the function/method with the mathematical operation(s) inside a given function. I still think with the given line numbers for a stack trace, it may be good enough to debug whatever issues may be needed. If you think this is still something that adds value, can you open a new issue? I would like to get this out of the door as is without too much more scope creep, unless you think otherwise